### PR TITLE
OBB: Encapsulate ray functions into OBB class

### PR DESCRIPTION
**Pull request by gkjohnson**
In the interest of encapsulating some of the bounding volume logic like #623 - I've moved the ray intersection functions over onto the OBB class.

cc @Desplandis

### CODE REVIEW

File: src/three/math/TileBoundingVolume.js [hunk 1]
```diff
@@ -73,10 +66,9 @@ export class TileBoundingVolume {
 		if ( obb ) {
 
 			// the obb transform contains no scale
-			_ray.copy( ray ).applyMatrix4( obb.inverseTransform );
-			if ( _ray.intersectBox( obb.box, _obbVec ) ) {
+			if ( obb.intersectsRay( ray, _obbVec ) ) {
 
-				obbDistSq = obb.box.containsPoint( _ray.origin ) ? 0 : _ray.origin.distanceToSquared( _obbVec );
+				obbDistSq = obb.box.containsPoint( ray.origin ) ? 0 : ray.origin.distanceToSquared( _obbVec );
```
Comment (gkjohnson) [inline src/three/math/TileBoundingVolume.js:71]: This this point had the same issue as the function pointed out in #623 - the transform scale wasn't accounted for.

File: src/three/math/OBB.js [hunk 2]
```diff
@@ -43,6 +44,32 @@ export class OBB {
 
 	}
 
+	// returns boolean indicating whether the ray has intersected the obb
+	intersectsRay( ray ) {
+
+		_ray.copy( ray ).applyMatrix4( this.inverseTransform );
+		return _ray.intersectsBox( this.box );
```
Comment (Desplandis) [inline src/three/math/OBB.js:51]: If you wish to follow `three` interface for `OBB#intersectsRay`, it should be of type `Ray -> boolean`. You could also remove code duplication.

```suggestion
		return _ray.intersectRay( this.box, _norm ) !== null;
```

File: src/three/math/TileBoundingVolume.js [hunk 3]
```diff
@@ -73,10 +66,9 @@ export class TileBoundingVolume {
 		if ( obb ) {
 
 			// the obb transform contains no scale
-			_ray.copy( ray ).applyMatrix4( obb.inverseTransform );
-			if ( _ray.intersectBox( obb.box, _obbVec ) ) {
+			if ( obb.intersectsRay( ray, _obbVec ) ) {
```
Comment (Desplandis) [inline src/three/math/TileBoundingVolume.js:69]: Shall be `intersectRay`, `intersectsRay` only have one parameter and  thus `_obbVec` components will never be assigned.

```suggestion
			if ( obb.intersectRay( ray, _obbVec ) ) {
```

File: src/three/math/TileBoundingVolume.js (hunk 1 above)
Comment (Desplandis) [reply src/three/math/TileBoundingVolume.js:71]: Thanks for fixing this issue too. =)

**Review by Desplandis — COMMENTED** [review]
Looks good except a single typo, thanks for fixing the issue with the projection. =)
I will try tomorrow with some of my "scaled" tilesets!

File: src/three/math/TileBoundingVolume.js (hunk 3 above)
Comment (gkjohnson) [reply src/three/math/TileBoundingVolume.js]: Oops, thank you!

File: src/three/math/OBB.js (hunk 2 above)
Comment (gkjohnson) [reply src/three/math/OBB.js:51]: > it should be of type Ray -> boolean

What do you mean by this? `Ray.intersectsBox` already [returns a boolean value](https://threejs.org/docs/?q=ray#api/en/math/Ray.intersectsBox)

Comment (Desplandis) [reply src/three/math/OBB.js:51]: My bad, I read `intersectBox`.

**Pull request merged**